src/datasources/cache/cache.router.ts

```diff
@@ -554,4 +564,49 @@ export class CacheRouter {
       `${args.method}_${args.params}`,
     );
   }
+
+  static getStakingDeploymentsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_DEPLOYMENTS_KEY, '');
+  }
+
+  static getStakingNetworkStatsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_NETWORK_STATS_KEY, '');
+  }
+
+  static getStakingDedicatedStakingStatsCacheDir(): CacheDir {
+    return new CacheDir(this.STAKING_DEDICATED_STAKING_STATS_KEY, '');
+  }
+
+  static getStakingPooledStakingStatsCacheDir(pool: `0x${string}`): CacheDir {
+    return new CacheDir(`${this.STAKING_POOLED_STAKING_STATS_KEY}_${pool}`, '');
+  }
+
+  static getStakingDefiVaultStatsCacheDir(args: {
+    chainId: string;
+    vault: `0x${string}`;
+  }): CacheDir {
+    return new CacheDir(
+      `${args.chainId}_${this.STAKING_DEFI_VAULT_STATS_KEY}_${args.vault}`,
+      '',
+    );
+  }
+
+  /**
+   * Calculate cache directory for staking stakes.
+   *
+   * Note: This function hashes the validators public keys to keep the
+   * cache key short and deterministic. The probability of collision is
+   * 0.00000271% for each 1,000,000 sets of validators public keys.
+   *
+   * @param validatorsPublicKeys - Array of validators public keys
+   * @returns {@link CacheDir} - Cache directory
+   */
+  static getStakingStakesCacheDir(
```

Comment:
To create a cache key for this, I've opted for hashing the `validatorsPublicKeys` to keep the key short (see function documentation), instead of simply concatenating the keys to build the key. Longer keys can harm Redis performance, and the amount of keys this function could receive is not bounded. Lmk if you think there's a better way to handle this.